ably/realtime_conn.go

```diff
@@ -880,6 +880,33 @@ func (c *Connection) reauthorize(arg connArgs) {
 	c.reconnect(arg)
 }
 
+func (c *Connection) onClientAuthorize(ctx context.Context, token *TokenDetails) {
+	switch c.State() {
+	case ConnectionStateConnected:
+		c.log().Verbosef("starting client-requested reauthorization with token: %+v", token)
+
+		changes := make(connStateChanges, 2)
+		{
+			off := c.internalEmitter.Once(ConnectionEventUpdate, changes.Receive)
+			defer off()
+		}
+		{
+			off := c.internalEmitter.Once(ConnectionEventFailed, changes.Receive)
+			defer off()
+		}
+
+		c.send(&protocolMessage{
+			Action: actionAuth,
+			Auth:   &authDetails{AccessToken: token.Token},
+		}, nil)
+
+		select {
+		case <-ctx.Done():
+		case <-changes:
```

Comment:
If this receives a failed event, should that not be passed back to the original caller of `Authorize`? Otherwise, why does `Authorize` need to block on this call?

Reply:
True. It makes sense and RTC8a3 makes it clear.

To avoid conflicts with #356, which builds on top of this, do you mind if I handle this there?

Reply:
Sure 👍

Reply:
See https://github.com/ably/ably-go/pull/356/commits/1b2142e58b49788fe236afad8b65e195b378bcf9.